src/java/us/kbase/templates/module_java_impl.vm.properties

```diff
@@ -59,64 +72,103 @@ public class ${java_module_name}Server extends JsonServerServlet {
-        long minLength = params.getMinLength();
-        if (minLength < 0)
-            throw new IllegalStateException("min_length parameter shouldn't be negative (" + minLength + ")");
-        
-        WorkspaceClient wc = new WorkspaceClient(new URL(this.wsUrl), authPart);
-        wc.setAuthAllowedForHttp(true);
-        ContigSet contigSet;
-        try {
-            contigSet = wc.getObjects(Arrays.asList(new ObjectIdentity().withRef(
-                    workspaceName + "/" + contigsetId))).get(0).getData().asClassInstance(ContigSet.class);
-        } catch (Exception ex) {
-            throw new IllegalStateException("Error loading original ContigSet object from workspace", ex);
+        /* Step 1 - Parse/examine the parameters and catch any errors
+         * It is important to check that parameters exist and are defined, and that nice error
+         * messages are returned to users.  Parameter values go through basic validation when
+         * defined in a Narrative App, but advanced users or other SDK developers can call
+         * this function directly, so validation is still important.
+         */
+        final String workspaceName = params.getWorkspaceName();
+        if (workspaceName == null || workspaceName.isEmpty()) {
+            throw new IllegalArgumentException(
+                "Parameter workspace_name is not set in input arguments");
+        }
+        final String assyRef = params.getAssemblyInputRef();
+        if (assyRef == null || assyRef.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "Parameter assembly_input_ref is not set in input arguments");
+        }
+        if (params.getMinLength() == null) {
+            throw new IllegalArgumentException(
+                    "Parameter min_length is not set in input arguments");
+        }
+        final long minLength = params.getMinLength();
+        if (minLength < 0) {
+            throw new IllegalArgumentException("min_length parameter cannot be negative (" +
+                    minLength + ")");
         }
         
-        System.out.println("Got ContigSet data.");
+        /* Step 2 - Download the input data as a Fasta file
+         * We can use the AssemblyUtils module to download a FASTA file from our Assembly data
+         * object. The return object gives us the path to the file that was created.
+         */
+        System.out.println("Downloading assembly data as FASTA file.");
+        final AssemblyUtilClient assyUtil = new AssemblyUtilClient(callbackURL, authPart);
+        /* Normally this is bad practice, but the callback server (which runs on the same machine
+         * as the docker container running the method) is http only
+         * TODO Should allow the clients to not require a token, even for auth required methods,
+         * since the callback server ignores the incoming token. No need to transmit the token
+         * here.
+         */
```

Comment:
It isn't, the requirement is imposed by the java compiled clients. That's why the TODO is there - we need to change the clients so that we aren't forced to include the token.